libs/cluster/Session/ClusterSession.cs

```diff
@@ -86,20 +86,10 @@ public bool ProcessClusterCommands(RespCommand command, ReadOnlySpan<byte> bufSp
                 }
                 else if (command == RespCommand.FAILOVER)
                 {
-                    if (!CheckACLAdminPermissions(bufSpan, count, out bool success))
```

Comment:
From what I can tell this is now checked in `ProcessBasicCommands`, which is better. Ideally none of the command executors should have to check for permissions. The flow should be Parsing -> Permission Checks -> Command Dispatch. That should give us more flexibility, including giving users access to just a subset of the cluster commands. `@admin` should just translate to a predefined command mask.

Reply:
That is correct, the intent is to check permissions in one place.  Subcommands currently require special handling (since they aren't parsed "up front"), but that's something that will ideally be fixed (I left a comment above about a suggestion I was given about how to persue that).

One clarifying statement though, assuming Garnet is looking to mimic Redis behavior, the idea that a session has `@admin` privileges is a misnomer.  Sessions have permission to run certain (sub-)commands, possibly further constrained by key filters.  ACL categories are just a convenient shorthand for sets of commands, with the exception of `@all` which includes commands not yet loaded.  Accordingly, something like `+@admin -failover` is possible and `FAILOVER` should be denied in that case.

Reply:
Good point. The special case of @admin might be removable and replaced with a category.